File: 1_Two_Sum/Java/Solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int pos1 = 0;
        int pos2 = 0;

        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                if (nums[i] + nums[j] == target) {
                    pos1 = i;
                    pos2 = j;
                    break;

                }
            }
        }

        return new int[] { pos1, pos2 };
    }

}

File: 1_Two_Sum/Python/Solution.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_indices = {}  # Dictionary to store indices of numbers
        
        for i, num in enumerate(nums):
            complement = target - num
            if complement in num_indices:
                return [num_indices[complement], i]
            num_indices[num] = i
        
        # If no solution is found, return an empty list
        return []
